app/views/features/new.html.erb

```diff
@@ -7,7 +7,16 @@
 
 <%= form_for :revision, :as => :revision, :url => url, :html => {:class => "form"} do |f| %>
 
-  <% if f.error_messages != "" %><div class="alert alert-danger"><%= f.error_messages :header_message => "Your note couldn't be saved." %></div><% end %>
+    <% if @revision.errors.any? %>
+      <div class="alert alert-danger">
+        <h2><%= pluralize(@revision.errors.count, "error") %> prohibited this map from being saved:</h2>
```

Comment:
Looks okay so far @stacytonui, quick question, should it be `map` or `revision`?
```suggestion
        <h2><%= pluralize(@revision.errors.count, "error") %> prohibited this revision from being saved:</h2>
```

Reply:
In this instance, i believe it should be `@revision` - but, perhaps we haven't created that instance variable? 

See how in the controller, we define `@revision` and `@node` -- so that should be OK... let's check the logs, maybe? 

https://github.com/publiclab/plots2/blob/6c51f2eb1297e6125d99a659cdebb9292c7e879f/app/controllers/features_controller.rb#L31-L64

Reply:
@RuthNjeri I missed that, thankyou!..let me apply the changes

Reply:
Ah, so the error in the [Travis log](https://travis-ci.com/github/publiclab/plots2/jobs/398471740#L2299) is:

```
ERROR["test_should_get_new_feature_form", #<Minitest::Reporters::Suite:0x000000000ef68250 @name="FeaturesControllerTest">, 34.55161182399999]
 test_should_get_new_feature_form#FeaturesControllerTest (34.55s)
Minitest::UnexpectedError:         ActionView::Template::Error: undefined method `errors' for nil:NilClass
            app/views/features/new.html.erb:10:in `block in _app_views_features_new_html_erb___1910048933803863612_125461940'
            app/views/features/new.html.erb:8:in `_app_views_features_new_html_erb___1910048933803863612_125461940'
            test/functional/features_controller_test.rb:45:in `block in <class:FeaturesControllerTest>'
```

It's failing on line 10 -- could it be that we failed ` if @node.valid? ` on line 39 of the controller, and so when we try to render these errors, the `@revision` variable doesn't exist yet? Maybe we should wrap this whole errors section in a conditional like, maybe:

```ruby
<% if defined?(@revision) %>
...
<% end %>
```

Want to try?

Reply:
In this case, if @revision.valid? is false.. that means that @revision.errors will be present and therefore using @revision in this form does make sense...

Reply:
@jywarren that totally makes sense, since @revision is only assigned when a @node is valid..

@stacytonui you can disregard the suggestion I made on Gitter for now and first try what @jywarren has suggested, it could be the main cause.

Reply:
@jywarren it has worked, thankyou!..thanks for the help @RuthNjeri